lib/screenplay/pa_messages/pa_message.ex

```diff
@@ -0,0 +1,63 @@
+defmodule Screenplay.PaMessages.PaMessage do
+  @moduledoc """
+  Represents a PA Message that will be retrieved by RTS to play audio in stations.
+  """
+
+  import Ecto.Query
+
+  alias Screenplay.Alerts.Cache, as: AlertsCache
+  alias Screenplay.Repo
+
+  use Ecto.Schema
+
+  @derive {Jason.Encoder, except: [:__meta__]}
+
+  schema "pa_message" do
+    field(:alert_id, :string)
+    field(:start_time_utc, :time)
+    field(:end_time_utc, :time)
+    field(:days_of_week, {:array, :integer})
+    field(:sign_ids, {:array, :string})
+    field(:priority, :integer)
+    field(:interval_in_minutes, :integer)
+    field(:visual_text, :string)
+    field(:audio_text, :string)
+    field(:paused, :boolean)
+    field(:saved, :boolean)
+    field(:message_type, :string)
+
+    timestamps(type: :utc_datetime)
+  end
+
+  def get_active_messages(now_utc \\ DateTime.utc_now()) do
+    now_adjusted =
+      now_utc
+      # Shift UTC to EST to account for possible Daylight Savings Time
+      |> DateTime.shift_zone!("America/New_York")
+      # Shift time back 3 hours to account for MBTA's 3am-3am service day
+      |> DateTime.add(-180, :minute)
+      # Shift back to UTC so time can be used in DB query
+      |> DateTime.shift_zone!("Etc/UTC")
+
+    day_of_week = Date.day_of_week(now_adjusted)
+
+    current_time = DateTime.to_time(now_adjusted)
+
+    {custom_messages, alert_messages} =
+      Repo.all(
+        from m in __MODULE__,
+          where:
+            (is_nil(m.start_time_utc) and is_nil(m.end_time_utc) and not is_nil(m.alert_id)) or
+              (^day_of_week in m.days_of_week and
+                 (m.start_time_utc <= ^current_time and m.end_time_utc >= ^current_time))
+      )
+      |> Enum.split_with(&is_nil(&1.alert_id))
+
+    alert_ids = AlertsCache.alert_ids()
+
+    alert_messages =
+      Enum.filter(alert_messages, fn %{alert_id: alert_id} -> alert_id in alert_ids end)
```

Comment:
Good idea. Still getting warmed up to how Ecto queries work. And I agree that the query string should grow _too_ large because we only list open alert IDs. That is currently at 97 and I've never seen it grow much larger than that.